src/python/pants/backend/python/goals/lockfile_test.py

```diff
@@ -41,19 +56,32 @@ def _generate(*, rule_runner: RuleRunner, use_pex: bool) -> str:
     )
     digest_contents = rule_runner.request(DigestContents, [result.digest])
     assert len(digest_contents) == 1
-    return digest_contents[0].content.decode()
+    content = digest_contents[0].content.decode()
+    if not use_pex:
+        return content
 
-
-def test_poetry_lockfile_generation() -> None:
-    rule_runner = RuleRunner(
-        rules=[
-            *lockfile_rules(),
-            *pex.rules(),
-            QueryRule(GenerateLockfileResult, [GeneratePythonLockfile]),
-        ]
+    pex_header = dedent(
+        f"""\
+        // This lockfile was autogenerated by Pants. To regenerate, run:
+        //
+        //    ./pants generate-lockfiles --resolve=test
+        //
+        // --- BEGIN PANTS LOCKFILE METADATA: DO NOT EDIT OR REMOVE ---
+        // {{
+        //   "version": 2,
+        //   "valid_for_interpreter_constraints": [],
```

Comment:
If the constraints change, that (potentially) invalidates the resolve results and the lock needs to be re-generated.

Reply:
Spot on. Thank you. I will add that tomorrow.